src/main/java/uk/gov/companieshouse/company/links/processor/InsolvencyStreamProcessor.java

```diff
@@ -1,31 +1,49 @@
 package uk.gov.companieshouse.company.links.processor;
 
+import static uk.gov.companieshouse.company.links.processor.ResponseHandler.handleResponse;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.kafka.support.KafkaHeaders;
 import org.springframework.messaging.Message;
 import org.springframework.messaging.MessageHeaders;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.company.CompanyProfile;
+import uk.gov.companieshouse.api.delta.Insolvency;
+import uk.gov.companieshouse.api.delta.InsolvencyDelta;
+import uk.gov.companieshouse.api.model.ApiResponse;
 import uk.gov.companieshouse.company.links.exception.RetryErrorException;
 import uk.gov.companieshouse.company.links.producer.InsolvencyStreamProducer;
+import uk.gov.companieshouse.company.links.service.CompanyProfileService;
 import uk.gov.companieshouse.delta.ChsDelta;
+import uk.gov.companieshouse.logging.Logger;
 
 
 @Component
 public class InsolvencyStreamProcessor {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(InsolvencyStreamProcessor.class);
+    private final Logger logger;
     private final InsolvencyStreamProducer insolvencyStreamProducer;
+    private final CompanyProfileService companyProfileService;
 
+    /**
+     * Construct an insolvency stream processor.
+     */
     @Autowired
-    public InsolvencyStreamProcessor(InsolvencyStreamProducer insolvencyStreamProducer) {
+    public InsolvencyStreamProcessor(InsolvencyStreamProducer insolvencyStreamProducer,
+                                     CompanyProfileService companyProfileService,
+                                     Logger logger) {
         this.insolvencyStreamProducer = insolvencyStreamProducer;
+        this.companyProfileService = companyProfileService;
+        this.logger = logger;
     }
 
     /**
-     * process.
+     * Process CHS Delta message.
      */
     //TODO What model should we use here? Is it a Avro?
     public void process(Message<ChsDelta> chsDelta) {
```

Comment:
Good spot thank you, made the change everywhere - (de)serialisers, processors, consumers, kafka configs and integration tests - also for charges as a placeholder so we're consistent until that is implemented with whatever object it may need. Repo no longer uses ChsDelta.